src/spark_rapids_ml/core.py

```diff
@@ -238,14 +238,14 @@ def initialize_cuml_logging(verbose: Optional[Union[bool, int]]) -> None:
             cuml_logger.set_level(log_level)
 
 
-class _CumlEstimator(Estimator, _CumlCommon, _CumlParams):
+class _CumlCaller(_CumlParams, _CumlCommon):
```

Comment:
maybe _CumlFitCaller?

Reply:
Yeah, mostly sparkcuml calls fit functions, but it also calls kneighbor that belongs to NearestNeighborModel class.  Maybe we keep _CumlCaller for now and decide if we should rename in the future.